x-pack/metricbeat/module/azure/client.go

```diff
@@ -88,12 +84,18 @@ func (client *Client) InitResources(fn mapMetric, report mb.ReporterV2) error {
 }
 
 // GetMetricValues returns the specified metric data points for the specified resource ID/namespace.
-func (client *Client) GetMetricValues(report mb.ReporterV2) error {
+func (client *Client) GetMetricValues(report mb.ReporterV2, metrics []Metric) []Metric {
+	var resultedMetrics []Metric
 	// loop over the set of metrics
-	for i, metric := range client.Resources.Metrics {
+	for _, metric := range metrics {
 		// select period to collect metrics, will double the interval value in order to retrieve any missing values
+		//if timegrain is larger than intervalx2 then interval will be assigned the timegrain value
```

Comment:
Just a suggestion here: wouldn't it be better to be calculated in a different helping method? :)